engine/schema/src/main/java/com/cloud/storage/dao/GuestOSHypervisorDaoImpl.java

```diff
@@ -100,6 +103,21 @@ public GuestOSHypervisorVO findByOsIdAndHypervisor(long guestOsId, String hyperv
         return findOneBy(sc);
     }
 
+    @Override
+    public GuestOSHypervisorVO findByOsIdAndHypervisor(long guestOsId, String hypervisorType, String hypervisorVersion) {
+        GuestOSHypervisorVO mapping = findByOsIdAndHypervisorInternal(guestOsId, hypervisorType, hypervisorVersion);
+        if (mapping != null || !Hypervisor.HypervisorType.VMware.toString().equals(hypervisorType)) {
+            return mapping;
+        }
+        String parentVersion = CloudStackVersion.getVMwareParentVersion(hypervisorVersion);
+        if (parentVersion == null) {
+            return null;
```

Comment:
Can we have a log message here stating that the parent is not found, so returning null.